### PR TITLE
logging: Added Redaction Filter for logs

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

logging: Added Redaction Filter for logs

This is adds the redaction filter for the logging mechanism. 
It is the first of many PRs to come, from the same branch.
- [x] Add Redaction Filter.
- [ ] Add Json Array Formatter.
- [ ] Add setup_logging function.
- [ ] Provide a way for non-root user to log.
- [ ] Migrate the older log files.

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Comment by orndorffgrant** [comment]
@CalvoM Thanks! Can we make this PR actually use the redaction filter and remove the old redaction method? That would make it a nice self-contained feature that we can test and release independently of the rest of the logging refactor if we need to.

File: uaclient/log.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    def redact_log(self, content: str) -> str:
+        if not isinstance(content, str):
+            content = str(content)
+        redacted_log = util.redact_sensitive_logs(content)
+        return redacted_log
+
+    def filter(self, record: logging.LogRecord):
+        record.msg = self.redact_log(record.msg)
+        return True
```
Comment (orndorffgrant) [inline uaclient/log.py:17]: Can these be combined into one function?

```suggestion
    def filter(self, record: logging.LogRecord):
        record.msg = util.redact_sensitive_logs(str(record.msg))
        return True
```

File: uaclient/util.py [hunk 2]
```diff
@@ -336,9 +334,9 @@ def readurl(
     debug_msg = "URL [{}] response: {}, headers: {{{}}}, data: {}".format(
         method or "GET", url, sorted_header_str, content
     )
-    if potentially_sensitive:
-        # For large responses, this is very slow (several minutes)
-        debug_msg = redact_sensitive_logs(debug_msg)
+    # if potentially_sensitive:
+    #     # For large responses, this is very slow (several minutes)
+    #     debug_msg = redact_sensitive_logs(debug_msg)
```
Comment (orndorffgrant) [inline uaclient/util.py:339]: I think we need a solution to this in this PR.

I believe `potentially_sensitive` is only set to `False` in one place - it defaults to true.

We can change the semantics of this to get a similar result though. The only place where this is set to `False` doesn't really need the full response to be logged anyway.

So maybe remove the `potentially_sensitive` argument, and add a new `log_response_body` argument that defaults to `True`. Then here we'd have something like:
```python
    debug_msg = "URL [{}] response: {}, headers: {{{}}}".format(
        method or "GET", url, sorted_header_str
    )
    if log_response_body:
        # For large responses, this is very slow (several minutes)
        debug_msg += ", data: {}".format(content)
    logging.debug(debug_msg)
```

**Review by orndorffgrant — COMMENTED** [review]
Looking good! Just two things I think need addressing. Also I haven't done any manual checks yet to verify that logs are still properly redacted.

File: uaclient/log.py (hunk 1 above)
Comment (CalvoM) [reply uaclient/log.py:17]: Oh yes, Ithought of this approach but then I noticed it fails when we pass the exception object to it. Let me confirm that it does not fail or if I can find a better way of logging it.

Comment (CalvoM) [reply uaclient/log.py:17]: Oh yes, thanks.

**Review by orndorffgrant — APPROVED** [review]
LGTM thanks!

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**